unreleased_history/new_features/tail_size_estimation_for_compaction.md

```diff
@@ -0,0 +1 @@
+Added tail size estimation during SST construction to prevent compaction files from exceeding their target size. Previously, compaction logic only considered data size when determining where to cut files, failing to reserve space for the tail (index, filter and other meta blocks) appended to the file. With this new feature enabled using the `compaction_use_tail_size_estimation` option, compaction files are now more accurately sized, reducing the likelihood of exceeding target sizes. This feature only supports some index and filter types. Refer to `compaction_use_tail_size_estimation` in DBOptions for more details.
```

Comment:
Here's my speech about release notes:

In short, people don't read them. To maximize chances of people actually reading them, we need to be extra judicious with their time and distill what we say down to what has changed in the user experience and what might break stuff downstream. And how to prepare for future changes. People can look at commit history if they want details.

Suggestion:
Added new option `target_file_size_is_upper_bound ` that makes most compaction output SST files come close to the target file size without exceeding it, rather than commonly exceeding it by some fraction (current behavior). For now the new behavior is off by default, but we expect to enable it by default in the future.